server/bin/pbench-index.py

```diff
@@ -381,6 +387,24 @@ def sigquit_handler(*args):
                         Path(INCOMING_rp, controller),
                     )
 
+                    try:
+                        dataset = Dataset.attach_or_create(
```

Comment:
Why use `attach_or_create()` here?  Should it always be `attach()`?

Reply:
`attach_or_create` allows for importing a previously unknown dataset when we're not sure whether it exists; it's what I'm currently using in indexing, for example. But it's a temporary convenience; in the long run we'll want `attach` which reconnects to an existing database and fails if somehow it doesn't exist... which is why I factored this out into a separate method layered on `attach`.